src/xtgeo/xyz/_polygons_oper.py

```diff
@@ -0,0 +1,284 @@
+
+* boundary_from_points (input to class method)
+* simplify_polygons (instance method)
+
+
+Functions starting with '_' are local helper functions
+"""
+from math import ceil
+
+import numpy as np
+import pandas as pd
+import shapely.geometry as sg
+from scipy.spatial import Delaunay, cKDTree
+
+import xtgeo
+
+xtg = xtgeo.XTGeoDialog()
+logger = xtg.functionlogger(__name__)
+
+
+MINIMUM_NUMBER_POINTS = 4
+
+
+class BoundaryError(ValueError):
+    """Get error on not able to create a boundary."""
+
+
+def boundary_from_points(points, alpha_factor=1.0, alpha=None, concave=False):
+    """From a Point instance, make boundary polygons (generic)."""
+    if not isinstance(points, xtgeo.Points):
+        raise ValueError("The input points is not an instance of xtgeo.Points")
+
+    if points.nrow < MINIMUM_NUMBER_POINTS:
+        logger.info("No. points is %s, need >= %s", points.nrow, MINIMUM_NUMBER_POINTS)
+        raise ValueError(
+            "Too few points to derive a boundary, present is "
+            f"{points.nrow}, need at least {MINIMUM_NUMBER_POINTS}"
+        )
+
+    if alpha_factor <= 0.0:
+        raise ValueError("The alpha_factor value must be greater than 0.0")
+
+    if alpha is not None and alpha <= 0.0:
+        raise ValueError("The alpha value must be greater than 0.0")
+
+    if concave:
+        alpha_factor = 0
+        alpha = 999  # dummy
+
+    usepoints = points.copy()  # make a copy since points may be filtered
+
+    xvec = usepoints.dataframe[usepoints.xname].values
+    yvec = usepoints.dataframe[usepoints.yname].values
+    zvec = usepoints.dataframe[usepoints.zname].values
+
+    if alpha is None:
+        # use scipy to detect average distance; 30 points should be sufficient
+        xyv = np.column_stack((xvec, yvec))
+        use_npoints = 30 if xvec.size >= 30 else xvec.size
+        kdtree = cKDTree(xyv)
+        dist, _ = kdtree.query(xyv, k=use_npoints)
+        auto_alpha = np.mean(dist)
+        logger.info("Proposed auto alpha is %s", auto_alpha)
+
+        # now try this auto_alpha, and iterate to a gradually larger alpha proposal
+        # if still no success in creating a boundary
+        try:
+            return_values = _create_boundary_polygon(
+                xvec, yvec, zvec, auto_alpha * alpha_factor
+            )
+        except BoundaryError as berr:
+            # iterate and propose a new alpha value while raising an exception
+            propose_alpha = _propose_new_alpha(
+                xvec, yvec, zvec, alpha_factor, auto_alpha
+            )
+            propose_alpha_factor = alpha_factor * propose_alpha / auto_alpha
+            msg = (
+                "Your alpha_factor or alpha is too low. Either set the alpha_factor to "
+                f"approx. {propose_alpha_factor:.3f}, or set alpha to approx. "
+                f"{propose_alpha}. Setting the alpha_factor (not alpha) is "
+                "recommended\n"
+            )
+            raise BoundaryError(msg) from berr
```

Comment:
I like this proposal function 👍 and seems to be working quite nicely for the alpha_factor. But got some strange behavior for the alpha.. the proposed value was very much dependent on the alpha_factor I tried to use. Example from my given pointset:
using alpha_factor 0.2 --> `proposed alpha=192` `proposed alpha_factor=0.355`
using alpha_factor 0.1 --> `proposed alpha=374` `proposed alpha_factor=0.346`
using alpha_factor 0.05 --> `proposed alpha=728` `proposed alpha_factor=0.337`

as it seems that the alpha_factor is quite stable, maybe we should drop proposing an alpha?

I also did not get any proposal when applying a too low **alpha** to the function, then I only got an BoundaryError. Not sure if this was intended